### PR TITLE
STYLE: Replace postfix by prefix increment in `for` loops in Core/Common

### PR DESCRIPTION
**Pull request by N-Dekker**
Followed common C++ guidelines, including:

Google C++ Style Guide: "Use the prefix form (++i) of the increment and
decrement operators unless you need postfix semantics."
https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement

Herb Sutter, Andrei Alexandrescu - C++ Coding Standards: 101 Rules,
Guidelines, and Best Practices: "Prefer the canonical form of ++ and --.
Prefer calling the prefix forms"
https://www.oreilly.com/library/view/c-coding-standards/0321113586/ch29.html

Applied at GNU bash, by the following command:

	find . \( -iname *.cxx -or -iname *.hxx -or -iname *.h \) -exec perl -pi -w -e 's/(\s+for \(.*; .+); (\w+)\+\+\)/$1; ++$2)/g;' {} \;

Manually replaced a few more postfix increments, which were not detected
by this regex command, as remarked by Mikhail Isakov (@issakomi).

### CODE REVIEW

File: Modules/Core/Common/include/itkArray.h [hunk 1]
```diff
@@ -104,7 +104,7 @@ class ITK_TEMPLATE_EXPORT Array : public vnl_vector<TValue>
   {
     this->m_LetArrayManageMemory = true;
     this->SetSize(r.GetSize());
-    for (SizeValueType i = 0; i < r.GetSize(); i++)
+    for (SizeValueType i = 0; i < r.GetSize(); ++i)
```
Comment (dzenanz) [inline Modules/Core/Common/include/itkArray.h:107]: Ahhh, `++i` looks so strange!

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Common/include/itkArray.h (hunk 1 above)
Comment (N-Dekker) [reply Modules/Core/Common/include/itkArray.h:107]: > Ahhh, `++i` looks so strange!

@dzenanz Are you serious? We could introduce a function, `itk::Increment(i)`...  😸

**Comment by N-Dekker** [comment]
FYI, it looks like a Clang-Tidy option to maintain this style is underway:
"[clang-tidy] Add performance-prefer-preincrement check"
https://reviews.llvm.org/D72553

**Review by dpshamonin — APPROVED** [review]

**Comment by dpshamonin** [comment]
:+1:

**Review by hjmjohnson — APPROVED** [review]

File: Modules/Core/Common/test/itkNeighborhoodIteratorTest.cxx [hunk 2]
```diff
@@ -83,7 +83,7 @@ itkNeighborhoodIteratorTest(int, char *[])
 
   it3.Print(std::cout);
   unsigned x, y, i;
-  for (y = 0, i = 0; y < 5; y++)
+  for (y = 0, i = 0; y < 5; ++y)
   {
     for (x = 0; x < 5; x++, i++)
```
Comment (issakomi) [inline Modules/Core/Common/test/itkNeighborhoodIteratorTest.cxx:88]: FYI, here post-increment was not detected by regex, probably (and several below)

Comment (dzenanz) [reply Modules/Core/Common/test/itkNeighborhoodIteratorTest.cxx:88]: Probably all multi-variable increments are missed.

Comment (N-Dekker) [reply Modules/Core/Common/test/itkNeighborhoodIteratorTest.cxx:88]: Good catch @issakomi Do you have a suggestion how to catch and fix such cases by regex? 

Obviously we must be 100% sure that the semantics don't change by the regex!

**Comment by N-Dekker** [comment]
Back to "work in progress": an update is underway!

File: Modules/Core/Common/test/itkNeighborhoodIteratorTest.cxx (hunk 2 above)
Comment (issakomi) [reply Modules/Core/Common/test/itkNeighborhoodIteratorTest.cxx:88]: Sorry, not yet sure how to automate this in reliable way. Looks like in Core/Common multi-variable _for_  with _postfix_ increment are below, at least with "_for_" and "_,_" at a single line

```
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkNeighborhoodIteratorTest.cxx:    for (x = 0; x < 5; x++, i++)
./test/itkObjectFactoryTest.cxx:    for (std::list<std::string>::const_iterator o = overrides.begin(); o != overrides.end(); ++o, ++n, ++d, e++)
./test/itkPriorityQueueTest.cxx:  for (; it != sequence.end(); ++it, i++)
./test/itkObjectFactoryTest2.cxx:      for (std::list<std::string>::const_iterator o = overrides.begin(); o != overrides.end(); ++o, ++n, ++d, e++)
./include/itkNeighborhoodIterator.hxx:    for (N_it = N.Begin(), this_it = this->Begin(); this_it < _end; this_it++, N_it++)
./include/itkNeighborhoodIterator.hxx:    for (N_it = N.Begin(), this_it = this->Begin(); this_it < _end; this_it++, N_it++)
./include/itkNeighborhoodIterator.hxx:    for (N_it = N.Begin(), this_it = this->Begin(); this_it < _end; N_it++, this_it++)
```

Comment (dzenanz) [reply Modules/Core/Common/test/itkNeighborhoodIteratorTest.cxx:88]: We don't have to change every instance now. Getting most of them is sufficient improvement.

Comment (N-Dekker) [reply Modules/Core/Common/test/itkNeighborhoodIteratorTest.cxx]: Please check my latest force-push 😄  https://github.com/InsightSoftwareConsortium/ITK/compare/ff9e1a7584564ff6a54b5191277847793ffe9fe4..d7eea8a76120648be23474c379894ef0338878f0

**Review by dzenanz — APPROVED** [review]

**Pull request merged**